### ISSUE TITLE
App crash on Drawer Menu Button Click in Release Mode

### ISSUE THREAD

**Issue by ravindraguptacapgemini** [react-navigation/react-navigation#11594]
### Current behavior

Run app on simulator as 
`npx react-native run-ios --mode Release`
When you click on menu button it crashes.

Crash log when running from Xcode in release mode

```
AwesomeProject_Latest[63788:766711] [native] [GESTURE HANDLER] Initialize gesture handler for view <RCTSurfaceView: 0x115e05fc0; frame = (0 0; 390 844); autoresize = W+H; gestureRecognizers = <NSArray: 0x600000c3cdb0>; layer = <CALayer: 0x6000002750e0>>
libc++abi: terminating due to uncaught exception of type facebook::jsi::JSError: Maximum call stack size exceeded.
```

### Expected behavior

When running app in release mode it should not crash on clicking on menu button.

### Reproduction

https://github.com/ravindraguptacapgemini/AwesomeProject_Latest/tree/react-navigation-drawer-crash

### Platform

- [X] Android
- [X] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [X] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [ ] @react-navigation/stack
- [ ] @react-navigation/native-stack
- [ ] react-native-tab-view

### Environment

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               |6.1.7|
| @react-navigation/drawer               |6.6.3|
| @react-navigation/stack                |6.3.17|
| react-native-safe-area-context         |4.7.2|
| react-native-screens                   |3.25.0|
| react-native-gesture-handler           |2.12.1|
| react-native-reanimated                |3.5.1|
| react-native                           |0.72.4|
| node                                   |16.20.0|
| npm                            |9.6.7|

**Comment by ravindraguptacapgemini** [comment]
Is there any issue with GitHub-actions, why marking 'needs repro' when the repro is already added to the issue, please check it https://github.com/ravindraguptacapgemini/AwesomeProject_Latest/tree/react-navigation-drawer-crash

**Comment by radko93** [comment]
As you noted in https://github.com/software-mansion/react-native-reanimated/issues/5075#issuecomment-1717591262, it's probably caused by reanimated. Can you try downgrading Reanimated version?

**Comment by kacperkapusciak** [comment]
Looks like duplicate of https://github.com/software-mansion/react-native-reanimated/issues/5067

Patch in React Native Reanimated is on it's way

**Comment by efstathiosntonas** [comment]
fixed on `react-native-reanimated@3.5.2`
